---
title: "STAMPS 2025: How to write a great application"
author: "Amy D Willis"
date: "February 21 2025"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{STAMPS 2025: How to write a great application}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# STAMPS @ the MBL

If you're reading this, I assume you know what [STAMPS](https://www.mbl.edu/education/advanced-research-training-courses/course-offerings/strategies-and-techniques-analyzing-microbial-population-structures) is and have some interest in applying -- but just in case, STAMPS is a ~10 day intensive training course offered at the Marine Biological Laboratory. Our curriculum is designed to help researchers analyze their microbiome data, focusing on bioinformatics and statistics skills and knowledge. It's primarily taught by methods developers for their target users, and we emphasize "getting your hands dirty" in the data. We also want you to understand the concepts underpinning the tools, so we don't teach click-button / copy-and-paste tools. Past participants of the course have described it as transformative for their science, highly impactful on their careers, and a ton of fun. If you have questions about the course or would like to learn more, please check out the [official](https://www.mbl.edu/education/advanced-research-training-courses/course-offerings/strategies-and-techniques-analyzing-microbial-population-structures) or [unofficial](https://mblstamps.github.io/) course website for more information. 

I'm co-Directing STAMPS 2025 at the MBL with Titus Brown again (our third year, woo!), and wanted to make some suggestions about how to write a fantastic STAMPS application. Applications are competitive, with only ~35 places, and I want you to write the best possible application.

# Background on applying

The most important thing is this: **if you think the course could benefit you, please apply**! Don't talk yourself out of applying just because it's competitive. 

In admissions, we prioritize "How much could this applicant benefit from the course?"  You could think of this as "Did the applicant make a strong case for why they need their seat?" 

There are three main questions on the application:

- Why do you want to take this course and why at this time? _Please describe how this course will impact your current research program as well as your long-term career goals. (up to 500 words)_
- What do you want us to know about yourself? _Please use this space to discuss the ways you feel you can contribute to this program and any unique aspects of your personal path as related to your passion and enthusiasm for science. (up to 500 words)_
- **Optional**: If you have other accomplishments or related information that is not reflected on your CV, please feel free to briefly describe them in the box below.

Questions 2 and 3 are important, but I'm going to focus on Question 1 in this post. 

# Structuring your application

Here is the basic structure of a great response to "Why do you want to take this course and why at this time?":

- "I am a [undergrad/MS candidate/PhD candidate/postdoc/PI]. I have been in that position for [X] years" or some less awkward way to tell us that you're a 1st vs 5th year :)  
- "The goal of my thesis/postdoc/research program is to study [big picture question]. I am specifically focusing on [more specific question/s]." 
	- Your application is being reviewed by a committee who have worked in the microbiome field for many years, so most applicants can be brief here. If your disease/host organism/biome is really unusual, feel free to include some additional background. 
- "I plan to analyze data from [this biome/cohort/study]", OR "I am still designing my experiment". Please go into methodological and experimental details here as much as you can! i.e., **What data do you have/want to have, who designed the experiment and collected the data, and when will you get it?** Here are some hypothetical examples: 
	- I am planning to collect my samples in June-August 2025, and perform 16S sequencing on them in September-October 2025... 
	- I have paired qPCR and RNASeq data from samples I obtained in 2023. I am planning to additionally collect shotgun sequencing data in November 2025... 
	- I am in the process of designing my experiment, and expect to have my samples by June 2027. I have not decided what observations I will have... 
	- I plan to analyze data collected by my colleague for my first project, and then design my own experiment and analyze it for my second project. The data that I have in hand is... and the data I plan to collect is... 
	- I arranged Sanger sequencing at a core facility on 20 *B fragilis* isolates collected in my lab. The data is with the core and we expect to receive it in February 2025...
	- and so on. (Yours will probably be very different to *all* of the above!)
- What computational, bioinformatic and/or statistical tools you intend to use or have used. Here some examples:
	- I have my data and I have no good ideas about what to do with it. I can open it in Excel... 
	- I have used R (reading in files and making basic plots) and the command line (including vim) for RNASeq before, but this is the first microbiome project of my research group and no one around me knows what to do with this type of data... 
	- I have started the DADA2 vignettes but I'm running into error messages that I can't solve. I have some basic R experience (using packages...) but have never used the command line... 
	- I have run metaphlan on my shotgun data and strainphlan on my isolate data and have loaded that into python and made some custom visualizations. I'd like to turn these into publication-ready figures and have specific questions about the assumptions made by these tools...
	- I have built a computational pipeline to train custom HMMs. I'd like to discuss the nuances of training sequence classification models with an expert... 

Participants in STAMPS 2024 came from the full spectrum of backgrounds. We had some very experienced users, people who had never seen a terminal before, and everything in between. And *everyone* learnt something!!!

So, think: "How can I convince them that I am ready to use what I learn in this course to advance my research?" 

# Misc

- If you applied in a previous year, and think the course would still benefit you, please reapply. Feel free to mention that you have applied in a previous year -- your sustained interest in attending the course will be viewed positively. 
- Feel free to tell us about your favourite papers' analyses! "I would like to learn how to build a custom BLAST database to search for X" or "I would like to learn how to make a [volcano plot/pangenome visualisation/heatmap of distances]". Show us that you are curious and motivated to learn! 
- In addition to specific tools and scientific/methodological questions that you have, feel free to tell us about other workshops (in person or online) that you have attended. 
- Not sure if the course is right for you? [Last year's schedule and slides](https://github.com/mblstamps/stamps2024/wiki#schedule) may help you get an idea of what we cover (we update the course every year, though!). 
- At this time, we don't have financial aid to offer. **If you would like to make a gift to support STAMPS participants, please contact me and Titus. Gifts can make this transformational learning opportunity more accessible!** 
- Dying to get started? [Happy Belly Bioinformatics](https://astrobiomike.github.io/) is my favorite online microbial bioinformatics resource. Anything you learn in advance of STAMPS will help you get more out of STAMPS.  
- Remember that learning is challenging -- your brain is rewiring itself to make new connections! Don't be discouraged if you feel overwhelmed by the task of analyzing your data. It might be is difficult and uncomfortable, but you can do it! Keep trying, and be kind to yourself. Don't be afraid to ask for help.  
- Questions? Feel free to email me and Titus (preferably both of us on the same email). 

I hope to see you in Woods Hole in July, and I look forward to reading your application! 
