<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>PNF: radEmu for differential abundance • blog</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PNF: radEmu for differential abundance">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">blog</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/pnf_extn_spaces.html">PNF: Extension spaces for phylogenetic trees</a></li>
    <li><a class="dropdown-item" href="../articles/pnf_radEmu.html">PNF: radEmu for differential abundance</a></li>
    <li><a class="dropdown-item" href="../articles/stamps_2025.html">STAMPS 2025: How to write a great application</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>PNF: radEmu for differential abundance</h1>
                        <h4 data-toc-skip class="author">Amy D
Willis</h4>
            
            <h4 data-toc-skip class="date">February 7 2025</h4>
      

      <div class="d-none name"><code>pnf_radEmu.Rmd</code></div>
    </div>

    
    
<p><a href="https://github.com/statdivlab/radEmu/" class="external-link">Software</a> | <a href="https://statdivlab.github.io/radEmu/" class="external-link">Documentation</a> | <a href="https://arxiv.org/abs/2402.05231" class="external-link">Preprint</a> | <a href="https://github.com/statdivlab/radEmu_supplementary" class="external-link">Supplementary</a></p>
<p><strong>TL;DR</strong>: Amy recommends radEmu for differential
abundance analysis. She recommends it above corncob, ALDEx2, regression
on compositional data transformations, anything in the ANCOM family…
basically above anything else that’s out there right now.</p>
<p>I don’t even know where to start with this paper. It’s just a great
method. I love everything about it. But this post is about why YOU
should love it. Hmmm, where should I start?</p>
<p>There’s a lot of rhetoric in the microbiome data analysis field. Is
microbiome data zero-inflated? Overdispersed? Negative Binomial?
Beta-Binomial? Compositional? Purple with pink spots?</p>
<p>In my view, these are the wrong questions.</p>
<p>Instead, I think we should be asking questions that are grounded in
our science:</p>
<ul>
<li>How many different bacterial species are there in polar oceans?
((breakaway)[] is good for that one)</li>
<li>Are there more different species in amended or unamended soils?
(betta might help)</li>
<li>What taxa are more abundant in gut microbiomes of folx diagnosed
with colorectal cancer than an age- and sex-matched non-cancer controls?
That’s RADEMU!</li>
</ul>
<p>One of the things I love most about radEmu is that is starts with
biology, not data. So, rather than “We fit fancy model X to the data,”
we start with “How much more abundant on average is A. bacterum in site
A than site B?”</p>
<div class="section level2">
<h2 id="if-you-take-the-blue-pill">If you take the blue pill…<a class="anchor" aria-label="anchor" href="#if-you-take-the-blue-pill"></a>
</h2>
<p>radEmu answers that question.</p>
<p>But don’t take the blue pill.</p>
</div>
<div class="section level2">
<h2 id="if-you-take-the-red-pill">If you take the red pill…<a class="anchor" aria-label="anchor" href="#if-you-take-the-red-pill"></a>
</h2>
<p>We showed that this question can’t be answered from HTS data under
reasonable assumptions (like “some taxa are consistently
over-detected”). This isn’t a radEmu problem – it’s a mathematical fact
about our data. I want to strongly emphasise that ALDEx2, ANCOM-BC2,
etc., all sweep this under the rug and don’t address this issue - not to
shame them, but to highlight the advantage of going the “scientific
question -&gt; can we answer it? -&gt; here’s a method” approach rather
than the “data has feature X -&gt; here’s a method that captures feature
X”.</p>
<p>So, what can we learn from HTS data? Well, we can answer “How much
more abundant on average is A. bacterium in site A than site B compared
to A. notherbacterium?” That’s cool, but most people don’t have A.
notherbacterium that they care about. Great news – we can answer “How
much more abundant on average is A. bacterium in site A than site B
compared to the average difference between taxa?” So, instead of
choosing a reference taxon, you can just compare it to the average. You
don’t have to assume that average change is zero, of course! You’re just
looking for what’s most different. Think of it this way: if a list of
fold-changes is 20, 1.2, 1.1, 1.1, 1, 0.9 and 0.1, you’re probably most
interested in the 20x and 0.1x fold-changers. 1.1 isn’t <em>no</em> fold
change, but compared to 20, it’s probably not that much.</p>
<p>At this point, people are often like, “Sounds good, so do you just do
some compositional transformation and t-test?” No. No, we don’t.
Instead, we set up an optimisation problem that guarantees a good
estimate of those (centered) fold-changes. So, I can’t given you a
general one-line closed form expression for it. But, you can sleep
soundly knowing that the estimate - is guaranteed to get more and more
right as you get more samples (experimental design implication:
prioritize more samples over deeper sequencing) - converges in
distribution to something nice = we can give you the p-values you
probably came for - doesn’t require you to “deal” with zeroes - doesn’t
require you to choose amongst data transformations (CLR, ALR, TSS…) -
doesn’t require you to choose a reference group - doesn’t assume that
your data is Poisson, Negative Binomial, Super Fancy But Arbitrary
Hierarchical Model etc-ed distributed - <strong>estimates something
real</strong> (actual fold-changes in abundance) under clear assumptions
(taxa may be overdetected; but consistently so)</p>
<p><em>Here’s a not-so-fun fact, by the way – “intuitive” methods that
log-transforms pseudocount-modified data are guaranteed to converge to
the wrong answer (“inconsistency”). It may not be very wrong in
practice, but it’ll still be wrong. Maybe that’s it’s own blog post for
another day.</em></p>
<p>Part of the reason we can side-step “zeroes and logs” is that our
optimisation problem models log(mean counts), not mean(log counts). As
the compositional zealots love to remind you, the latter isn’t defined
if you ever have zeroes. But, then… what does it mean to talk about the
average of a collection of numbers with a non-zero probability of an
infinite value? It doesn’t mean anything, in my opinion, which is one
more reason not to do it that way. Instead, as long as there is a
positive probability of the species being detected in some environment,
log(mean counts) is well-defined.</p>
<p>Oh yeah, and our optimisation problem doesn’t care about counts
(integers) or coverages (non-negative numbers). It’s all the same to us.
So, give radEmu your metagenomic data! Nom nom.</p>
<p>And those are the main ideas! The rest is just details, really. We
don’t simply do a t-test to give you p-values because it turns out that
the p-values are too small. We don’t even do a fancy “robust” t-test,
because they’re still too small. We want you to have the best, most
perfect p-values possible (and we love these details), so our p-values
come from robust score tests, which performed beautifully in practice.
Great work, co-author David!</p>
<p>[A world where Amy knows how to insert memes. The meme is from
Christopher Nolan’s first batman movie. Christian Bale: “Am I supposed
to understand that?” Morgan Freeman: “No, but I want you to know how
hard it was.”]</p>
<p>Other fun facts - If you have a spike-in, great! Use that as your
reference group and you don’t have to worry about this centering thing.
- Yes, radEmu “can handle” longitudinal data. I put this in quotes
because it’s a question I get asked a lot, but a vacuous question
because there are a lot of different things you “can do” with
longitudinal data. radEmu can probably do what you want. Use the
<code>cluster</code> argument and check out <a href="https://statdivlab.github.io/radEmu/articles/radEmu_clustered_data.html" class="external-link">this</a>
tutorial.</p>
<p>Ok, interpreting radEmu output. Why don’t we talk about this more?!
It is so important. A corncob coefficient estimate of 0.5 means
something totally different to an ALDEx2 estimate of 0.5, which no one
seems to talk about! Anyway, here’s how you interpret a radEmu estimate
of 0.5 when using the default settings:</p>
<p>“We estimate that A. bacterium is 1.648 times more abundant in
[environment A] than in [otherwise similar] / [ environment B ] compared
to typical fold-changes in taxon abundances.”</p>
<p>1.648? That’s just exp(0.5). And the “otherwise similar” is important
if you’re adjusting for other variables – here’s another option:</p>
<p>“We estimate that A. bacterium is 1.648 times more abundant in
[environment A] than in [ environment B ]’s of the same temperature and
season, when compared to typical fold-changes in taxon abundances.”</p>
<p>It gets a bit wordy, but it’s very important to be clear what
comparison you’re making. (You don’t have to worry about “otherwise
similar” if you’re not adjusting for other variables.)</p>
<p>Also, If you’re finding radEmu to be too slow, let us know! We can
help you tune it, and it helps us to know how people are using the
method.</p>
<p><strong>This has not yet been done but will be done soon (Feb
2025):</strong> We updated the preprint to make the intro clearer, and
to add some more benchmarking – great work, co-author Sarah! I find the
benchmarking extremely convincing that radEmu has the best power out of
methods that control T1E rates. (And you shouldn’t even consider using
methods that don’t control T1E rates.)</p>
<p>radEmu works great up to… maybe 2000 taxa? If you have more taxa, you
might want our fast approximation method. So, watch this space.
<code>fastEmu</code> will have a friend here shortly…</p>
<p>Wrap-up: I really believe that radEmu is a great way to do
differential abundance, and because of that, I want you to use it. So,
please check it out, and let us know your feedback. If there’s something
we can do to convince you to use it, tell us, please!</p>
<p>If you have questions about the method, feel free to email me – I
might write another blog post about radEmu if more things come up.</p>
<p>Oh yeah, and radEmu is our affectionate nickname for radEmuAbPill.
“using <strong>r</strong>elative <strong>a</strong>bundance
<strong>d</strong>ata to <strong>e</strong>stimate
<strong>mu</strong>ltiplicative differences in <strong>ab</strong>solute
abundances with <strong>p</strong>artially <strong>i</strong>dentified
<strong>l</strong>og-<strong>l</strong>inear models.” Hope that
clarifies!</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Amy Willis, Sarah Teichman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
